src/NCalc/Parser/LogicalExpressionParser.cs

```diff
@@ -0,0 +1,301 @@
+#nullable disable
+
+using System;
+using System.Globalization;
+using NCalc.Domain;
+using NCalc.Exceptions;
+using Parlot.Fluent;
+using static Parlot.Fluent.Parsers;
+using Identifier = NCalc.Domain.Identifier;
+
+namespace NCalc.Parser;
+
+public static class LogicalExpressionParser
+{
+    private static readonly Parser<LogicalExpression> Parser;
+
+    private static readonly ValueExpression True = new(true);
+    private static readonly ValueExpression False = new(false);
+
+    static LogicalExpressionParser()
+    {
+        /*
+         * Grammar:
+         * expression     => ternary ( ( "-" | "+" ) ternary )* ;
+         * ternary        => equality ( "?" equality ":" equality)?
+         * equality       => relational ( ( "=" | "!=" | ... ) relational )* ;
+         * relational     => multiplicative ( ( ">=" | ">" | ... ) multiplicative )* ;
+         * multiplicative => unary ( ( "/" | "*" ) unary )* ;
+         * unary          => ( "-" | "not" ) unary
+         *                 | primary ;
+         *
+         * primary        => NUMBER
+         *                  | STRING
+         *                  | "true"
+         *                  | "false"
+         *                  | "[" anything "]"
+         *                  | function
+         *                  | "(" expression ")" ;
+         *
+         * function       => Identifier "(" arguments ")"
+         * arguments      => expression ( "," expression )*
+         */
+        // The Deferred helper creates a parser that can be referenced by others before it is defined
+        var expression = Deferred<LogicalExpression>();
+
+        var intParser = Terms.Integer().Then<LogicalExpression>(static d => new ValueExpression(d));
+
+        var intExponentParser = Terms.Integer(NumberOptions.AllowSign)
+            .And(Terms.Text("e", true))
+            .And(Terms.Integer(NumberOptions.AllowSign))
+            .Then<LogicalExpression>((context, d) =>
+            {
+                if (!string.IsNullOrWhiteSpace(d.Item2))
```

Comment:
I've reviewed it and if `(!string.IsNullOrWhiteSpace(d.Item2))` condition should be removed as we're using `And` and this one will be triggered only when it's specified